lib/linguist/heuristics.yml

```
@@ -360,6 +360,14 @@ disambiguations:
   rules:
   - language: Unity3D Asset
     pattern: 'tag:unity3d.com'
+- extensions: ['.mc']
+  rules:
+  - language: Win32 Message File
+    pattern: '(?i)MessageId=|^;'
```

I'd restrict this pattern so `MessageId=` only matches at the beginning of lines (possibly with `/*` sandwiched in-between):

```suggestion
    pattern: '(?i)^[ \t]*(?>\/\*\s*)?MessageId=|^;'
```

As for the `^;` pattern… recalling that M4 is a language-agnostic templating engine, it's possible that `;` might match in less common scenarios (say, when generating the source of a configuration file). I'd therefore match `M4`'s heuristic first, before `Win32 Message Files`. See below.